eo-runtime/src/test/eo/org/eolang/memory-tests.eo

```diff
@@ -88,3 +95,124 @@
       b.write 20
       a
     $.equal-to 10
+
+[] > memory-is-strictly-typed-int-error
+  memory 42 > m
+  nop > @
+    assert-that
+      try
+        []
+          m.write 3.1416 > @
+        [e]
+          e > @
+        nop
+      $.equal-to "Memory type missmatch: expected <int>, got <float>"
```

Comment:
@Graur Let's imagine, that `int` and `float` have the same size - 4 bytes (like in Java). Should we check their types? Or we might compare just size?

Reply:
@volodya-lombrozo Yes, you are right. Thanks! 
Moreover there is no chance to compare types if I try to implement it in EO. So we decided to avoid this changes